File: php/services/promotionService.php
<?php
class PromotionService {
    private $promotionRepository;

    public function __construct(PromotionRepository $promotionRepository) {
        $this->promotionRepository = $promotionRepository;
    }

    public function getAllPromotions(): array {
        return $this->promotionRepository->findAll();
    }

    public function getPromotionById(int $id): ?Promotion {
        return $this->promotionRepository->findById($id);
    }

}

File: src/main/java/com/example/web/services/postgres/PromotionService.java
package com.example.web.services.postgres;

import com.example.web.models.Promotion;
import com.example.web.repositories.postgres.PromotionRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class PromotionService {
    private final PromotionRepository promotionRepository;
    @Autowired
    public PromotionService(PromotionRepository promotionRepository){
        this.promotionRepository = promotionRepository;
    }
    public List<Promotion> findAll(){
        return promotionRepository.findAll();
    }
    public Optional<Promotion> findById(Integer id){
        return promotionRepository.findById(id);
    }
}
